pkg/util/fxutil/logging/sender.go

```diff
@@ -0,0 +1,211 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package logging
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"math/rand"
+	"net/http"
+	"os"
+	"runtime"
+	"strings"
+	"time"
+)
+
+// Span represents a Datadog APM span in JSON format for the v0.3/traces endpoint.
+type Span struct {
+	Service  string             `json:"service"`
+	Name     string             `json:"name"`
+	Resource string             `json:"resource"`
+	TraceID  uint64             `json:"trace_id"`
+	SpanID   uint64             `json:"span_id"`
+	ParentID uint64             `json:"parent_id"`
+	Start    int64              `json:"start"`
+	Duration int64              `json:"duration"`
+	Error    int32              `json:"error"`
+	Meta     map[string]string  `json:"meta,omitempty"`
+	Metrics  map[string]float64 `json:"metrics,omitempty"`
+	Type     string             `json:"type,omitempty"`
+}
+
+// sendSpansToDatadog sends an array of TraceSpan to the local trace-agent.
+// It uses the v0.3/traces endpoint with JSON format (array of traces).
+func (l *fxTracingLogger) sendSpansToDatadog(flavor string) {
+	l.mu.Lock()
+
+	globalMeta := map[string]string{"flavor": flavor}
+
+	// Generate a single trace ID for all spans
+	traceID := rand.Uint64()
+
+	// Generate span IDs for hierarchy
+	appSpanID := rand.Uint64()
+	constructorsSpanID := rand.Uint64()
+	onStartHooksSpanID := rand.Uint64()
+
+	// isErr := l.err != nil
+
+	// Create the parent lifecycle span
+	appSpan := &Span{
+		Service:  serviceName,
+		Name:     "Fx Application Startup",
+		Resource: "Fx Application Startup",
+		TraceID:  traceID,
+		SpanID:   appSpanID,
+		ParentID: 0,
+		Start:    l.startTime.UnixNano(),
+		Duration: l.endTime.Sub(l.startTime).Nanoseconds(),
+		Error:    errToCode(nil),
+		Type:     "custom",
+		Meta:     globalMeta,
+	}
+
+	// Create constructors aggregator span if we have constructors
+	constructorsAggSpan := &Span{
+		Service:  serviceName,
+		Name:     "Fx Constructors",
+		Resource: "Fx Constructors",
+		TraceID:  traceID,
+		SpanID:   constructorsSpanID,
+		ParentID: appSpanID,
+		Start:    l.startTime.UnixNano(),
+		Duration: l.lifecycleStart.Sub(l.startTime).Nanoseconds(),
+		Type:     "custom",
+		Meta:     globalMeta,
+	}
+
+	// Create onstart hooks aggregator span if we have onstart hooks
+	onStartAggSpan := &Span{
+		Service:  serviceName,
+		Name:     "Fx OnStart Hooks",
+		Resource: "Fx OnStart Hooks",
+		TraceID:  traceID,
+		SpanID:   onStartHooksSpanID,
+		ParentID: appSpanID,
+		Start:    l.lifecycleStart.UnixNano(),
+		Duration: l.endTime.Sub(l.lifecycleStart).Nanoseconds(),
+		Type:     "custom",
+		Meta:     globalMeta,
+	}
+	spans := make([]*Span, 0, len(l.spans)+2)
```

Comment:
You're adding 3 spans ?
```suggestion
	spans := make([]*Span, 0, len(l.spans)+3)
```